### PR TITLE
RunAsGroup fixes and tests

### PR DESCRIPTION
**Pull request by runcom**
**- What I did**

same as #1849 + a new test job for features :)

~~need to address @mrunalp comment here https://github.com/kubernetes-sigs/cri-o/pull/1849#issuecomment-430832689 as well but waiting on CI first~~

This PR also needs:

- ~~https://github.com/openshift/aos-cd-jobs/pull/1644~~
- ~~https://github.com/openshift/release/pull/2101~~
- ~~https://github.com/openshift/test-infra/pull/178~~

~~After the above are merged, we need to run the new `/test e2e_features` jobs~~

this is needed for 1.11 and 1.12 as well

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: contrib/test/integration/e2e-features.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            --test_args="-host=https://{{ ansible_default_ipv4.address }}:6443
+                        --ginkgo.focus=\[Feature:SecurityContext\]
+                        --ginkgo.skip=\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|PersistentVolumes|\[HPA\]|should.support.building.a.client.with.a.CSR|should.support.inline.execution.and.attach|should.propagate.mounts.to.the.host|for.NodePort.service|type.clusterIP|unready.pods|ExternalName.services|Guestbook.application|in-cluster.config|Pods.should.support.pod.readiness.gates
+                        --report-dir={{ artifacts }}"
+            &> {{ artifacts }}/e2e.log
+  # Fix vim syntax hilighting: "
+
+- block:
+
+    - name: Disable selinux during e2e tests
```
Comment (runcom) [inline contrib/test/integration/e2e-features.yml]: re-enable for security context tests? just a memo to myself :)

**Comment by giuseppe** [comment]
could this be a fix for https://github.com/kubernetes-sigs/cri-o/issues/1904?  Investigating that issue, it seems the place where it can happen is the same code you are heavily refactoring as part of this PR

**Comment by runcom** [comment]
> could this be a fix for #1904? Investigating that issue, it seems the place where it can happen is the same code you are heavily refactoring as part of this PR

not sure, that looks like a flake happening somewhere else as well (maybe even in selinux libraries on host and/or kernel :confused: )

**Comment by runcom** [comment]
Need to re enable selinux, hold on

**Comment by runcom** [comment]
@mrunalp PTAL, I've also changed the code to call into `getExecUser` just once as you suggested in https://github.com/kubernetes-sigs/cri-o/pull/1849#issuecomment-430832689

**Comment by kolyshkin** [comment]
> this is needed for 1.11 and 1.12 as well

Better late than never :) 1.11 backport is at https://github.com/cri-o/cri-o/pull/4115

**Pull request merged**